core/src/main/kotlin/com/expediagroup/sdk/core/client/OkHttpEventListener.kt

```diff
@@ -19,41 +19,54 @@ import com.expediagroup.sdk.core.constant.HeaderKey
 import com.expediagroup.sdk.core.plugin.logging.ExpediaGroupLoggerFactory
 import okhttp3.Call
 import okhttp3.Connection
-import okhttp3.EventListener
 import okhttp3.Handshake
 import okhttp3.Protocol
 import okhttp3.Request
 import okhttp3.Response
+import okhttp3.EventListener
 import java.io.IOException
 import java.net.InetSocketAddress
 import java.net.Proxy
+import java.util.UUID
+import java.util.concurrent.atomic.AtomicReference
+
+data class OkHttpEventListener private constructor(private val transactionId: AtomicReference<UUID>) : EventListener() {
 
-object OkHttpEventListener : EventListener() {
     private val log = ExpediaGroupLoggerFactory.getLogger(this::class.java)
 
-    fun Call.getTransactionId() = request().headers[HeaderKey.TRANSACTION_ID]
+    // Factory for creating EventListeners with transaction IDs
+    companion object {
+        // Expose the factory as a static property
+        val FACTORY: EventListener.Factory = object : EventListener.Factory {
+            override fun create(call: Call): EventListener {
+                val transactionIdHeader = call.request().header(HeaderKey.TRANSACTION_ID)
+                val transactionId = AtomicReference(UUID.fromString(transactionIdHeader))
+                return OkHttpEventListener(transactionId)
+            }
+        }
+    }
 
     override fun callStart(call: Call) {
         super.callStart(call)
-        log.debug("Call start for transaction-id: [${call.getTransactionId()}]")
+        log.debug("Call start for transaction-id: [${transactionId.get()}]")
```

Comment:
```suggestion
        log.debug("Call start for transaction-id: [{}]", transactionId.get())
```
Consider using parameterized logging (`log.debug("... [{}]", transactionId.get())`) instead of string interpolation. This approach is more efficient as it avoids string construction when debug level is disabled.

This applies to all instances in the file.